zulipterminal/ui_tools/views.py

```diff
@@ -1492,6 +1493,55 @@ def keypress(self, size: urwid_Size, key: str) -> str:
         return super().keypress(size, key)
 
 
+class TopicInfoView(PopUpView):
+    def __init__(self, controller: Any, stream_id: int, topic: str) -> None:
+        self.stream_id = stream_id
+        self.controller = controller
+        stream = controller.model.stream_dict[stream_id]
+        self.topic_name = topic
+        stream_name = stream["name"]
+
+        title = f"{stream_name} {STREAM_TOPIC_SEPARATOR} {self.topic_name}"
+
+        topic_info_content = [
+            ("Topic settings", []),
+        ]  # type: PopUpViewTableContent
+
+        popup_width, column_widths = self.calculate_table_widths(
+            topic_info_content, len(title)
+        )
+
+        resolve_topic_setting = urwid.CheckBox(
+            label="Topic Resolved",
+            state=self.topic_name.startswith(RESOLVED_TOPIC_PREFIX),
+            checked_symbol=CHECK_MARK,
+        )
+        urwid.connect_signal(
+            resolve_topic_setting, "change", self.toggle_resolve_status
+        )
+        popup_width = max(
+            popup_width,
+            len(resolve_topic_setting.label) + 4,
+        )
+
+        self.widgets = self.make_table_with_categories(
+            topic_info_content, column_widths
+        )
+
+        self.widgets.append(resolve_topic_setting)
+        super().__init__(controller, self.widgets, "TOPIC_DESC", popup_width, title)
+
+    def toggle_resolve_status(self, button: Any, new_state: bool) -> None:
+        if self.topic_name.startswith(RESOLVED_TOPIC_PREFIX):
+            self.controller.model.toggle_topic_resolved_status(
+                self.stream_id, self.topic_name
+            )
+        else:
+            self.controller.model.toggle_topic_resolved_status(
+                self.stream_id, RESOLVED_TOPIC_PREFIX + self.topic_name
```

Comment:
It seems strange to have the model function called `toggle_...` which you then pass a different parameter to depending upon the topic name, and which is the *new* name - other than relying on the called function to do the toggling. That is, I'd expect a toggle function to be passed in fixed parameters. In other words, a toggle is a 'move to the other position', not a 'move to *this* position'.

For at least some similar cases we have logic in the model function instead. Take a look at the muting and pinning logic, for example.